CONTRIBUTING.md

```diff
@@ -0,0 +1,122 @@
+* Annotations on method arguments shall, to the largest extent possible, be on the same line as the method argument (and, if present, before the `final` modifier)
+
+### Javadoc
+
+* Each class shall have a Javadoc comment.
+* Unless the method is `private`, it shall have a Javadoc comment.
+* Getters and Setters shall have a Javadoc comment. The following is prefered
+    ```
+    /**
+     * The count of widgets
+     */
+    private int widgetCount;
+    
+    /**
+     * Returns the count of widgets.
+     *
+     * @return the count of widgets. 
+     */
+    public int getWidgetCount() {
+        return widgetCount;
+    }
+    
+    /**
+     * Sets the count of widgets.
+     *
+     * @param widgetCount the count of widgets.
+     */
+    public void setWidgetCount(int widgetCount) {
+        this.widgetCount = widgetCount;
+    }
+    ```
+* When adding a new class / interface / etc, it shall have a `@since` doc comment. The version shall be `FIXME` (or `TODO`) to indicate that the person merging the change should replace the `FIXME` with the next release version number. The fields and methods within a class/interface (but not nested classes) will be assumed to have the `@since` annotation of their class/interface unless a different `@since` annotation is present.
+
+### IDE Configuration
+
+* Eclipse, by and large the IDE defaults are acceptable with the following changes:
+    - Tab policy to `Spaces only`
+    - Indent statements within `switch` body
+    - Maximum line width `120`
+    - Line wrapping, ensure all to `wrap where necessary`
+    - Organize imports alphabetically, no grouping
+* NetBeans, by and large the IDE defaults are acceptable with the following changes:
+    - Tabs and Indents
+        + Change Right Margin to `120`
+        + Indent case statements in switch
+    - Wrapping
+        + Change all the `Never` values to `If Long`
+        + Select the checkbox for Wrap After Assignement Operators
+* IntelliJ, by and large the IDE defaults are acceptable with the following changes:
+    - Wrapping and Braces
+        + Change `Do not wrap` to `Wrap if long`
+        + Change `Do not force` to `Always`
+    - Javadoc
+        + Disable generating `<p/>` on empty lines
+    - Imports
+        + Class count to use import with '*': `9999`
+        + Names count to use static import with '*': `99999`
+        + Import Layout
+            * import all other imports
+            * blank line
+            * import static all other imports
+    
+## Issues
+
+This project uses [Jenkins Jira issue tracker](https://issues.jenkins-ci.org) 
+with [github-plugin](https://issues.jenkins-ci.org/browse/JENKINS/component/15896) component.
+            
+## Links 
+
+- https://wiki.jenkins-ci.org/display/JENKINS/contributing
+- https://wiki.jenkins-ci.org/display/JENKINS/Extend+Jenkins
+- https://wiki.jenkins-ci.org/display/JENKINS/GitHub+commit+messages
```

Comment:
@jtnord included his earlier "GitHub+commit+messages" into "contributing" page. So this link can be removed now